### ISSUE TITLE
Missing pagination controls

### ISSUE THREAD

**Issue by dekhaus** [marmelab/ng-admin#506]
Hi

I have a working ng-admin site.  I'm writing today because for some reason I have yet to figure out - my list view is missing the pagination controls.  The information is there - aka the "1 - 10 of 20" - just no pagination controls.  

Reviewing the documentation - I realized that my server wasn't adding the X-Total-Count key/value to the response header.  I added that hoping that would fix the problem - but it didn't. 

I checked the DOM on my running instance of the ng-admin-demo and noticed the na-datagrid-pagination directive and the corresponding data.  

Reviewing the page generated by my app - I only see an empty na-datagrid-pagination directive.

any idea(s) what's going on.

Thanks
Dave

**Comment by dekhaus** [comment]
This afternoon - I did some more testing trying to isolate the issue and made an interesting discovery.  First some context ... my ng-admin application (codename Rocket) currently is managing two remote resources, let's call them Products and Categories (both managed by the same server - RemoteServer).  Let's assume that Rocket is served by LocalServer.

Rocket has no problem rendering the list views for both Products and Categories.  I've confirmed that the RemoteServer is adding the X-Total-Count header to any 'index' responses.

The problem is that Rocket's list view for remote resources is missing the pagination controls.

What I discovered was that when I add a local resource to Rocket (a resource called Place) - Rocket's Places list view DOES contain pagination controls.

In short - remote resources do not have pagination controls whereas local resources do have pagination controls.  I've confirmed that all response headers have the correct X-Total-Count value.

Any help would be greatly appreciated.

Thanks
Dave

**Comment by fzaninotto** [comment]
That's extremely weird. Did you define a custom `perPage()` on your entity? Also, since we `parseInt` the value contained in the `X-Total-Count` header, is it correctly formatted (e.g. no leading 0)? Lastly, do you have a Restangular interceptor altering the request or the response?

**Comment by dekhaus** [comment]
Hi.

   When defining my listView - I did call the perPage method.  

   Any listView 'backed' by a remote resource has an incorrect value for the total number of records.  The value displayed is the value of the perPage argument.

   I have lots of sample data and deployed apps to share with you to illustrate the problem.   

   First - I deployed a Places endpoint to Heroku ( https://daves-places.herokuapp.com/places ).  This endpoint respects the default pagination parameters used by ng-admin (_page and _perPage) as well as the pagination parameters used in the ng-admin-demo (_start and _end).   The source code for Places endpoint is here -> https://github.com/dekhaus/places

   Second - using the Places endpoint - I deployed another a Rails 4 app called Recipes to Heroku ( https://daves-recipes.herokuapp.com ).  The source code for Recipes is here ( https://github.com/dekhaus/Recipes ).  Recipes manages two resources ...

    The first resource Recipes manages is a local one called Recipes.  By 'local' I mean the 'recipes' endpoint is ( https://daves-recipes.herokuapp.com/recipes ) - aka part of the same Rails app.  The pagination for the Recipes listView works perfectly.

    The second resource Recipes manages is a remote one called Places (the aforementioned Places endpoint -> https://daves-places.herokuapp.com/places).  The pagination for Places is not correct and exhibits the bug we've been discussing - specifically the missing pagination controls and the incorrect total amount.  If you view the Recipes 'places' listView ( https://daves-recipes.herokuapp.com/#/places/list ) and examine the network call to the 'places' endpoint ( https://daves-places.herokuapp.com/places?_page=1&_perPage=5&_sortDir=DESC&_sortField=id ) - you can clearly see that while the X-Total-Count has the correct value (200) - the pagination is not functioning correctly for 'remote' resources.

    I would happily debug this issue - but I only have access to the minified ng-admin source.  There's another issue that you responded to regarding debugging ng-admin - but I'm too new to AngularJS/JS to understand your instructions on how to generate the non-minified ng-admin source.  If you could provided detailed instructions I could then debug this issue.

    I also modified the ng-admin-demo to manage the remote Places endpoint.  The results are the same as above.  I created this pastie (http://pastie.org/10238118) in case you want to run the code locally.

    I spent a lot of time today (Friday) dealing with this issue - trying everything I could to prove that I didn't do something stupid.  Hopefully the information I provided makes it easy for you to replicate the issue / bug and if I'm lucky provide a fix.

Thanks
Dave

**Comment by nglekhoi** [comment]
Maybe you should add the header "Access-Control-Expose-Headers" with the value "x-total-count".
It worked for me.
My WebAPI code for reference :
```
HttpContext.Current.Response.AppendHeader("Access-Control-Expose-Headers", "x-total-count");
```

**Comment by dekhaus** [comment]
IT WORKED !!!!  You are a LIFESAVER.
